x-pack/plugin/identity-provider/src/main/java/org/elasticsearch/xpack/idp/action/TransportPutSamlServiceProviderAction.java

```diff
@@ -56,6 +59,10 @@ protected void doExecute(Task task, final PutSamlServiceProviderRequest request,
             listener.onFailure(new IllegalArgumentException("request document must not have an id [" + document.docId + "]"));
             return;
         }
+        if (document.nameIdFormat != null && identityProvider.getAllowedNameIdFormats().contains(document.nameIdFormat) == false) {
```

Comment:
I don't like this much as this is _yet another_ part of the code where we do some validation on the Document but this is the only place that validation depends on IDP configuration so it didn't make sense for this to be part of SecurityProviderDocument. I'm happy to discuss or take suggestions for refactoring this.

Reply:
I think this makes sense. We don't want to fail writes to documents just because the nameID format isn't valid - for example, it would make it impossible to disable a broken SP.

It would be logically sensible to do it in the request, but requests aren't really setup to do that sort of validation.